src/hidra/receiver/datareceiver.py

```diff
@@ -259,6 +260,97 @@ def stop(self):
         self.run_loop = False
 
 
+def run_plugin_thread(
+        plugin_name, plugin_config, target_dir, data_queue, log, event):
+    # Execute all plugin code on a separate thread to protect main thread from
+    # slow or blocking plugins
+    try:
+        plugin_m = import_module("plugins." + plugin_name)
+        plugin = plugin_m.Plugin(plugin_config)
+        plugin.setup()
+        log.info("Loading '%s' plugin", plugin_name)
+    except Exception:
+        log.error("Could not load '%s' plugin", plugin_name, exc_info=True)
+        return
+
+    while not event.is_set():
+        try:
+            data = data_queue.get(timeout=0.5)
+        except queue.Empty:
+            continue
+
+        try:
+            [metadata, data] = data
```

Comment:
is the rhs object 'data' immutable? otherwise this is asking for trouble

Reply:
It is mutable but should not be used after processing. Added comment to clarify that.